### ISSUE TITLE
Revive development for this version

### ISSUE THREAD

**Issue by toby63** [mumble-voip/mumble-iphoneos#129]
As you can see this version for iOS was last updated in September 2017.
The version number is confusing (1.3.1 in [Appstore](https://apps.apple.com/us/app/mumble/id443472808)).

The main developer is busy/inactive for now, so maybe someone wants to take over the development of this app.
https://github.com/mumble-voip/mumble-www/issues/96#issuecomment-639209986

If you are interested, post a comment :slightly_smiling_face:.

**Comment by AToska21** [comment]
I could have a go at trying to update it

**Comment by toby63** [comment]
@AToska21 Feel free to open pull requests or start discussions, outlining what you plan to do.
You can also get into contact with the developers directly, see for example here: https://www.mumble.info/contribute/#getting-started

**Comment by Krzmbrzl** [comment]
> Could you pin this issue maybe? Just a small idea.

What for? It's not like this issue gets lost in more recent reports.

I don't have any contact to mkrautz. So I can't really do much more than pinging him here as well :shrug: 

@AToska21 That'd be much appreciated :+1:

**Comment by toby63** [comment]
>What for? It's not like this issue gets lost in more recent reports.

I could also ask why not? It is the central topic for this repo imo.

> I don't have any contact to mkrautz. So I can't really do much more than pinging him here as well :shrug:

I would assume that someone in the team has it.

**Comment by Kissaki** [comment]
>  The version number is confusing (1.3.1 in Appstore).

Confusing how?

---

I agree with Krzmbrzl. Pinning this ticket as it is doesn’t add much. If we want to clarify this project is up for adoption/seeking maintainers, we should do so more explicitly both in the README and pin a/this ticket with a more explicit title (like “Seeking maintainers”).

---

@AToska21 It would be great if you could work on this project, either one-off updates to make it work in current dev envs and end-user runtimes, or towards maintaining it.

**Comment by toby63** [comment]
>> The version number is confusing (1.3.1 in Appstore).

> Confusing how?

It might misslead users into believing that version 1.3.1 is equal to the official version 1.3.1, while it is not.

**Comment by toby63** [comment]
I changed the title a bit, so maybe we could ping this now.
If you can provide more useful information, that would be great too.

**Comment by vierzwei** [comment]
I could also take a look at maintaining this app. 

I created an iOS App that also integrates MumbleKit (https://apps.apple.com/de/app/yourtalk-fm/id1171446553#?platform=iphone) so I understand the basic structure of the app. The thing is, that the app needs a lot of rewrite even to make it possible to publish an update. It would be more like re-starting than maintaining.

**Comment by Kissaki** [comment]
That would be great @vierzwei 

Could you elaborate on what needs to be done and why? What are the big stepping stones/milestones?

If you are willing to commit setting up tickets and milestones, or a ticket with a todo list may be a good idea.

**Comment by Kissaki** [comment]
@AToska21 we have not heard back from you since your initial interest. Does that still stand? Are you able to commit some time? Do you have prior knowledge and an idea of what needs to be done too?

**Comment by Kissaki** [comment]
> And could you maybe talk to @mkrautz ?; I know he does not have much time, but maybe he could give a little bit of information what probably needs to be done, where new devs should start etc.

This is very unlikely to happen. We are better off going on without waiting here.

**Comment by zDEFz** [comment]
Since we are still in the COVID-19 phase, and mumble appears to be one of the only working solutions to jam over the Internet, a development here would be much appreciated.

Especially if you want to use mumble, but theres no M1 support, so you use your smartphone, to also get Screenreader / VoiceOver support...

**Comment by Sleuth56** [comment]
Sense this repo looks like it's maintained once more it would be good to remove the part in the readme that references to his thread.

**Comment by davidebeatrici** [comment]
It's still unmaintained.

**Comment by toby63** [comment]
@davidebeatrici @Krzmbrzl  @Kissaki Just an idea: maybe you could add a note about the search for new maintainers/team members on the website or at least also in the main repo, so more people see it :thinking:.
Other projects such as the LibreOffice Team even showed a message in their app (even though that is maybe a bit too much).

**Comment by Krzmbrzl** [comment]
I don't think a new maintainer is found by putting that on the website. Code contributors usually come if they have a personal interest in contributing and if they do they'll have a look at the source code at which point they will probably notice this issue.

**Comment by zDEFz** [comment]
> I don't think a new maintainer is found by putting that on the website. Code contributors usually come if they have a personal interest in contributing and if they do they'll have a look at the source code at which point they will probably notice this issue.

Or just implement web-mumble with websockets and access through the browser... Its more of a pain though.

**Comment by toby63** [comment]
> I don't think a new maintainer is found by putting that on the website. Code contributors usually come if they have a personal interest in contributing and if they do they'll have a look at the source code at which point they will probably notice this issue.

I disagree.
There are a lot of programmers out there and some of them will also use your software.
How should they know that this project needs help?
A Note is exactly what might get people's attention.

**Comment by seniorm0ment** [comment]
It would be very nice to see either a new Mumble app for iOS, or polish up the old one and maintain it. It is definitely quite dated, and missing a handful of features such as local volume adjustment according to some people. I was just talkikg with a friend and he was on iOS, honestly was surprised it went as smoothly as it did seeing the app has been unmaintained for so long.

**Comment by mkrautz** [comment]
Hello.

My Apple Developer Account expires in January.

Pinging this to see if anyone would like to take over the app. I can transfer the app to another Apple Developer account if so.

As mentioned, I don't have the time to maintain this, as is obvious by the lack of commits. But if anyone would like to take over, please let me know.

**Comment by carsoncall** [comment]
Hello! @mkrautz 

My name is Carson, and I am a software developer and senior CS student. My friends and I use Mumble all of the time, and I am interested in taking over as a maintainer. I would like to modernize the app as well. I don’t have a ton of experience contributing to FOSS, but I am passionate about using it and would like to take this on.

**Comment by Krzmbrzl** [comment]
@carsoncall thanks for stepping up!
I would suggest you start by forking this repository and then contribute PRs. I will skim the first ones and if they seem reasonable I'll merge them and eventually give you direct write access to this repository.

**Comment by carsoncall** [comment]
Sounds good! Any suggestions on where to start? Is there a style guide etc for PRs? My buddy is a career Flutter dev, and the Android Mumble app is mostly unmaintained as well. Additionally, there already exists a Mumble client library in Dart (Dumble) that is only somewhat stale. We were thinking that, long term, this could be a good way to reduce maintenance costs for both platforms while also ensuring a consistent experience. What do you think?

**Comment by habnabit** [comment]
@carsoncall could you start by even just building+releasing a version of this for modern iOS targets?

**Comment by Kissaki** [comment]
The project still uses the "legacy build system" so it can't be built with XCode version 14. Upgrading the project to the new build system would be a great first task.

* 59b7b147b5dff67f003b8d28e8ede3452a1c0fe7
* https://github.com/mumble-voip/mumble-iphoneos/issues/131#issuecomment-1995615473

**Comment by Flofus** [comment]
@Kissaki why we don't build the app from scratch new or is it too much work?

**Comment by Krzmbrzl** [comment]
Well, if there is someone who wants to rewrite the app from scratch and actually does it, we would likely accept a PR. However, since we (the core Mumble devs) don't even have the time or knowledge to _maintain_ the existing app, we are certainly not capable of _rewriting_ it :shrug:

**Comment by chrissands** [comment]
I feel like you're going to get a lot more requests soon...

**Comment by Anuril** [comment]
The currently published app runs from iOS 8.0, but modernizing the applications requires the use of APIs which are only available from around ios 13. It would be great to know if we can agree on a version cut-off like ios 12

**Comment by Krzmbrzl** [comment]
@fridtjof I'll leave the decision up to you. Personally, I'd be fine with cutting support for older versions in favor of development making some progress again...

**Comment by Anuril** [comment]
Ios 11 received the last update in 2018. (See [here](https://en.wikipedia.org/wiki/IOS_version_history) ). I doubt it can even do secure TLS anymore, but I for sure can't check. My M1 Macbook can't install pre iOS15 simulators and I don't have devices with these old Versions. 

I have a lot of respect for the original writers of this application that it survived since the release 8 years ago and still somehow works, but at some point...

**Comment by Krzmbrzl** [comment]
Yeah, I'm totally with you on this. Assume that it is fine to wipe support for ancient iOS versions unless people with iOS development experience have really good arguments against it 👀

**Comment by Anuril** [comment]
> Yeah, I'm totally with you on this. Assume that it is fine to wipe support for ancient iOS versions unless people with iOS development experience have really good arguments against it 👀 

I mean my PR should build / run on iOS12 but I currently can't check soo... 😅 

I tried running on simulated iOS 15, but xcode downloads the simulator but doesn't allow me to select it 🤷

**Comment by fridtjof** [comment]
Yeah, iOS 13 seems reasonable. I think it's what i wanted to go for anyway last time i worked on this.

I've been meaning to get a (beta) release out half a year ago, but i've unfortunately been busy with life and other obligations so I kind of forgot 😅 I think there were also some issues I wanted to address before a full release, most importantly #179 and less urgently #161.

**Comment by Anuril** [comment]
> Yeah, iOS 13 seems reasonable. I think it's what i wanted to go for anyway last time i worked on this.
> 
> I've been meaning to get a (beta) release out half a year ago, but i've unfortunately been busy with life and other obligations so I kind of forgot 😅 I think there were also some issues I wanted to address before a full release, most importantly #179 and less urgently #161.

#161 is fixed with my pr, there's even a setting where you can re-enable notifications and get directed to the settings app after you initially declined

I could not verify #179, it didn't happen to me, so I didn't spend any time

**Comment by ssdrive-flaktra** [comment]
> > Yeah, I'm totally with you on this. Assume that it is fine to wipe support for ancient iOS versions unless people with iOS development experience have really good arguments against it 👀
> 
> I mean my PR should build / run on iOS12 but I currently can't check soo... 😅
> 
> I tried running on simulated iOS 15, but xcode downloads the simulator but doesn't allow me to select it 🤷

I have a few ancient iPhones with ~iOS 12-16. If you need testing I could give it a shot.

**Comment by zavitax** [comment]
Hey so I submitted PR https://github.com/mumble-voip/mumble-iphoneos/pull/185 which adds auto-retry in case of failure to connect or connection break. Hopefully you guys will find use for it...

**Comment by ogmkp** [comment]
Can't find Mumble today on the App Store, even though I installed it on my old iPhone. What's going on?

**Comment by mkrautz** [comment]
The developer account lapsed. I’ve renewed it. Should show up again soon, pending Apple delays in processing.

**Comment by ogmkp** [comment]
Thanks so much for your quick response! The appstore web page has just reappeared, but the app isn't showing up on appstore on phone, I'll wait a little longer.

**Comment by mkrautz** [comment]
It's available now, but apparently not ranking highly in App Store search - but that's probably unrelated. :)

**Comment by ogmkp** [comment]
The app now appears at the top of the search results in the App Store on the phone. This is important for me because it's the only app that let me avoid using private services with closed source. Thanks for your work.
